.github/workflows/build.yml

```diff
@@ -40,6 +43,48 @@ jobs:
             name: gfruit-image
             path: "./buck-out/v2/gen/root/**/*"
 
+  cosmo_seq:
+    needs: changes
+    if: ${{ needs.changes.outputs.buck2 == 'true'  || needs.changes.outputs.cosmo_seq == 'true' }}
+    runs-on: self-hosted
+    steps:
+        - run: echo "The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
+        - name: Check out repository code
+          uses: actions/checkout@v4
+          with:
+            submodules: 'true'
+        - name: Update pip reqs
+          run : python3 -m pip install --upgrade -r tools/requirements.txt --break-system-packages
+        - name: buck path
+          run: echo "~/.cargo/bin:/opt/Xilinx/Vivado/2024.1/bin" >> "$GITHUB_PATH"
+        - name: Build cosmo_seq bitstream
+          run: buck2 build //hdl/projects/cosmo_seq:cosmo_seq --show-output
+        - uses: actions/upload-artifact@v4
+          with:
+            name: cosmo-seq-image
+            path: "./buck-out/v2/gen/root/**/*"
+
+  cosmo_hp:
+    needs: changes
+    if: ${{ needs.changes.outputs.buck2 == 'true' ||  needs.changes.outputs.cosmo_hp == 'true' }}
+    runs-on: self-hosted
+    steps:
+        - run: echo "The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
+        - name: Check out repository code
+          uses: actions/checkout@v4
+          with:
+            submodules: 'true'
+        - name: Update pip reqs
+          run : python3 -m pip install --upgrade -r tools/requirements.txt --break-system-packages
+        - name: buck path
+          run: echo "~/.cargo/bin:/opt/Xilinx/Vivado/2024.1/bin:/opt/oss-cad-suite-20240513/bin" >> "$GITHUB_PATH"
+        - name: Build cosmo_hp bitstream
+          run: buck2 build //hdl/projects/cosmo_hp:cosmo_hp_bitstream --show-output
```

Comment:
nit: this is the only target with `_bitstream` in the name. I'd prefer we add it to `grapefruit` and `cosmo_seq` as well or remove it here and have the target name just be `cosmo_hp`.